ambry-api/src/main/java/com/github/ambry/frontend/s3/S3MessagePayload.java

```diff
@@ -351,4 +353,144 @@ public String toString() {
       return "Bucket=" + bucket + ", Key=" + key + ", UploadId=" + uploadId;
     }
   }
+
+  public static class S3BatchDeleteObjects {
+
+    // Ensure that the "Delete" wrapper element is mapped correctly to the list of "Object" elements
+    @JacksonXmlElementWrapper(useWrapping = false)  // Avoids wrapping the <Delete> element itself
+    @JacksonXmlProperty(localName = "Object")      // Specifies that each <Object> element maps to an instance of S3BatchDeleteKeys
+    private List<S3BatchDeleteKey> objects;
+
+    public List<S3BatchDeleteKey> getObjects() {
+      return objects;
+    }
+
+    public void setObjects(List<S3BatchDeleteKey> objects) {
+      this.objects = objects;
+    }
+
+    @Override
+    public String toString() {
+      return "S3BatchDeleteObjects{" +
+          "objects=" + objects +
+          '}';
+    }
+  }
+
+  public static class S3BatchDeleteKey {
+
+    // Maps the <Key> element inside each <Object> to the 'key' property in S3BatchDeleteKeys
+    @JacksonXmlProperty(localName = "Key")
+    private String key;
+
+    public String getKey() {
+      return key;
+    }
+
+    public void setKey(String key) {
+      this.key = key;
+    }
+
+    @Override
+    public String toString() {
+      return "S3BatchDeleteKey{" +
+          "key='" + key + '\'' +
+          '}';
+    }
+  }
+
+  // exact naming from S3BatchDelete response
+  public static class DeleteResult {
+
+    @JacksonXmlElementWrapper(useWrapping = false)
+    @JacksonXmlProperty(localName = "Error") // Maps to <Error> in XML
+    private ConcurrentLinkedQueue<S3MessagePayload.S3ErrorObject> errors;
+
+    @JacksonXmlElementWrapper(useWrapping = false)
+    @JacksonXmlProperty(localName = "Deleted") // Maps to <Deleted> in XML
+    private ConcurrentLinkedQueue<S3MessagePayload.S3DeletedObject> deleted;
+
```

Comment:
These object is just for serialization and deserialization, we don't need to use concurrent data structure for them.
Could you please use just ArrayList, and convert a concurrent linked queue to array list in setErrors and setDeleted?